src/js/utils/captions.js

```diff
@@ -0,0 +1,89 @@
+// NOTE: This is generic for any CSS/html list representation. Consider renaming and moving to generic module.
```

Comment:
Since these functions use "functional programming paradigms" (such as higher order functions and some branching polymorphism), I'll plan on adding some more code comments/descriptions here before final merge.